crates/turborepo-ui/src/tui/app.rs

```diff
@@ -255,35 +429,236 @@ fn update(
             app.next();
         }
         Event::ScrollUp => {
-            app.scroll(Direction::Up);
+            app.has_user_scrolled = true;
+            app.tasks
+                .get_mut(&app.active_task())
+                .unwrap()
+                .scroll(Direction::Up)
+                .unwrap_or_default();
         }
         Event::ScrollDown => {
-            app.scroll(Direction::Down);
+            app.has_user_scrolled = true;
+            app.tasks
+                .get_mut(&app.active_task())
+                .unwrap()
+                .scroll(Direction::Down)
+                .unwrap_or_default();
         }
         Event::EnterInteractive => {
-            app.interact(true);
+            app.has_user_scrolled = true;
+            app.interact();
         }
         Event::ExitInteractive => {
-            app.interact(false);
+            app.has_user_scrolled = true;
+            app.interact();
         }
         Event::Input { bytes } => {
             app.forward_input(&bytes)?;
         }
-        Event::SetStdin { task, stdin } => {
-            app.pane.insert_stdin(&task, Some(stdin))?;
+        Event::SetStdin { stdin } => {
+            app.insert_stdin(Some(stdin))?;
         }
         Event::UpdateTasks { tasks } => {
             app.update_tasks(tasks);
-            app.table.tick();
+            // app.table.tick();
         }
     }
     Ok(None)
 }
 
-fn view<I>(app: &mut App<I>, f: &mut Frame) {
-    let (_, width) = app.term_size();
-    let vertical = Layout::horizontal([Constraint::Fill(1), Constraint::Length(width)]);
-    let [table, pane] = vertical.areas(f.size());
-    app.table.stateful_render(f, table);
-    f.render_widget(&app.pane, pane);
+fn view(app: &mut App<Box<dyn io::Write + Send>>, f: &mut Frame, rows: u16, cols: u16) {
+    let horizontal = Layout::horizontal([Constraint::Fill(1), Constraint::Length(cols)]);
+    let [table, pane] = horizontal.areas(f.size());
+
+    let active_task = app.active_task();
+
+    let pane_to_render: TerminalPane<Box<dyn io::Write + Send>> = TerminalPane::new(
```

Comment:
We can omit the type hint here since `TerminalPane<W>` will have the same `W` as contained in `App`
```suggestion
    let pane_to_render = TerminalPane::new(
```